server/src/main/java/io/crate/auth/CachingJwkProvider.java

```diff
@@ -132,32 +88,35 @@ static URL urlForDomain(String domain) {
     }
 
     @Override
-    public Jwk get(String keyId) {
-        return cache.get(keyId, this::getResult).jwk;
+    public Jwk get(String keyId) throws JwkException {
+        var cache = result;
+        if (cache == null || cache.expired(clock.instant())) {
+            cache = getKeys();
+            result = cache;
+        }
```

Comment:
I think `JwkResult` needs a flag `cacheable` instead of relying on `cache.expired` working with `Duraiton.ZERO`.

It's not only semantically confusing but also can cache even if header is not provided?

ie, say we have a result with `clock.instant().plus(ttl) == clock.instant().plus(Duration.ZERO) -> clock.instant()` holding time of the very first response. Every next request will have clock.instant() > instant_of_the_first_request?

Reply:
>can cache even if header is not provided?

I was wrong, it won't

> Added a fixup to only cache if it's not expired.

👍